src/main/java/core/basesyntax/FigureSupplier.java

```diff
@@ -0,0 +1,67 @@
+package core.basesyntax;
+
+import core.basesyntax.figures.AbstractFigure;
+import core.basesyntax.figures.Circle;
+import core.basesyntax.figures.IsoscelesTrapezoid;
+import core.basesyntax.figures.Rectangle;
+import core.basesyntax.figures.RightTriangle;
+import core.basesyntax.figures.Square;
+import java.util.Random;
+
+public class FigureSupplier {
+    private static final int MAX_SIZE_VALUE = 101;
+    private static final String DEFAULT_COLOR = "white";
+    private static final int DEFAULT_NUMBER = 10;
+    private Random random = new Random();
+    private ColorSupplier colorSupplier = new ColorSupplier();
+
+    public AbstractFigure getRandomFigure() {
+        int index = random.nextInt(Figure.values().length);
+        String nameOfFigure = Figure.values()[index].name();
+
+        switch (nameOfFigure) {
+            case "CIRCLE" : return getRandomCircle();
+            case "ISOSCELES_TRAPEZOID" : return getRandomIsoscelesTrapezoid();
+            case "RECTANGLE" : return getRandomRectangle();
+            case "RIGHT_TRIANGLE" : return getRandomRightTriangle();
+            default : return getRandomSquare();
+        }
+    }
+
+    private AbstractFigure getRandomIsoscelesTrapezoid() {
+        IsoscelesTrapezoid isoscelesTrapezoid = new IsoscelesTrapezoid(
+                colorSupplier.getRandomColor(),
+                getRandomSize(), getRandomSize(), getRandomSize());
+        return isoscelesTrapezoid;

```

Comment:
In situations like this, there is no need to store the object in a local variable if it will be immediately used. Instead, you can pass the expression directly without assigning it to a variable